lib/transport.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const build = require('pino-abstract-transport')
+const SonicBoom = require('sonic-boom')
+const { defaults } = require('./utils')
+const { messageBuilderFactory: rfc5424 } = require('./rfc5424.js')
+const { messageBuilderFactory: rfc3164 } = require('./rfc3164.js')
+
+module.exports = pinoTransport
+
+function pinoTransport (options) {
+  const opts = Object.assign({}, defaults, options)
+
+  const processMessage = opts.modern ? rfc5424(opts) : rfc3164(opts)
+
+  const destination = getStream(opts.destination)
+
+  return build(function (source) {
+    source.on('data', function (obj) {
+      const canContinue = destination.write(processMessage(obj))
+      if (!canContinue) {
+        source.pause()
+        destination.once('drain', () => { source.resume() })
+      }
+    })
+  }, {
+    // close (err, cb) {
```

Comment:
It seems that the `.end` callback is never executed

Reply:
Why not?

Reply:
If I uncomment this section, the tests are failing

I need to track down why this is happening

Reply:
The issues was cause by the `end` usage.
When the destination is a `sonic-boom` stream, the `end` method does not accept any callback
https://github.com/pinojs/sonic-boom#sonicboomend

Instead, when a writable stream is set (stdout or stderr) the `end` consumes the callback correctly